feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,118 @@
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._base_transformers.base_numerical import BaseNumericalTransformer
+from feature_engine._base_transformers.mixins import (
+    FitFromDictMixin,
+    GetFeatureNamesOutMixin,
+)
+from feature_engine._check_input_parameters.check_init_input_params import (
+    _check_param_drop_original,
+)
+
+
+class DistanceFeatures(
+    BaseNumericalTransformer, FitFromDictMixin, GetFeatureNamesOutMixin
+):
+    EARTH_RADIUS: float = 6371.  # radius of Earth in kms
+
+    def __init__(
+            self,
+            a_latitude: str,
+            a_longitude: str,
+            b_latitude: str,
+            b_longitude: str,
+            output_column_name: Union[str, None] = None,
+            drop_original: bool = False,
+    ) -> None:
+
+        self.a_latitude = self._check_column_name(a_latitude)
+        self.a_longitude = self._check_column_name(a_longitude)
+        self.b_latitude = self._check_column_name(b_latitude)
+        self.b_longitude = self._check_column_name(b_longitude)
+
+        self.output_column_name = self._check_column_name(column_name=output_column_name)
+
+        _check_param_drop_original(drop_original=drop_original)
+        self.drop_original = drop_original
+
+        self.variables = None
```

Comment:
I don't think we need this parameter. I'd suggest using RelativeFeatures as template to model this class: https://github.com/VascoSch92/feature_engine/blob/e1e927625678ee73c5c3a9edcf79e955ff9c5e8e/feature_engine/creation/relative_features.py

Reply:
variables is a parameter that we have in all transformers, so I would stick to this name instead of using `coordinate_columns`

In short, let's replace coordinate_columns by variables.